modules/lvm2/udiskslinuxvolumegroup.c

```diff
@@ -784,13 +784,19 @@ wait_for_logical_volume_path (UDisksLinuxVolumeGroupObject  *group_object,
 }
 
 /* ---------------------------------------------------------------------------------------------------- */
+
+enum VolumeType { VOL_PLAIN, VOL_THIN_POOL, VOL_THIN_VOLUME };
+
 static gboolean
-handle_create_volume (UDisksVolumeGroup     *_group,
-                      GDBusMethodInvocation *invocation,
-                      const gchar           *arg_name,
-                      guint64                arg_size,
-                      GVariant              *options,
-                      gboolean               is_thin)
+handle_create_volume (UDisksVolumeGroup              *_group,
+                      GDBusMethodInvocation          *invocation,
+                      const gchar                    *arg_name,
+                      guint64                         arg_size,
+                      GVariant                       *options,
+                      const gchar                    *auth_error_msg,
+                      UDisksThreadedJobFunc           create_function,
+                      enum VolumeType                 vol_creation_type,
+                      const gchar                     *arg_pool)
```

Comment:
Again, I think as many as possible values from above should be determined/derived from ``vol_creation_type``.